extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryRepository.java

```diff
@@ -25,14 +25,16 @@
 import org.apache.rya.streams.api.entity.StreamsQuery;
 import org.apache.rya.streams.api.exception.RyaStreamsException;
 
+import com.google.common.util.concurrent.Service;
+
 import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * Repository for adding, deleting, and listing active queries in Rya Streams.
  */
 @DefaultAnnotation(NonNull.class)
-public interface QueryRepository extends AutoCloseable {
+public interface QueryRepository extends Service {
```

Comment:
Maybe indicate you only need to start this service if you're using the subscription methods. Might be a good idea to have those methods fail with an illegal state exception if the service isn't running.

Reply:
done